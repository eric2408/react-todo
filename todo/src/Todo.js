import React, {useState} from "react";
import UpdateTodoForm from "./UpdateTodoForm";

function Todo({task = "empty task", id = "1", remove, update}) {
    const [edit, setEdit] = useState(false);

    const handleRemove = ()  => {
        remove(id);
    }

    const handleUpdate = () => {
        setEdit(edit => !edit)
    };

    let jsx = (
        <div>
            <li key={id}>{task}</li>
            <button onClick={handleUpdate}>Update</button>
            <button onClick={handleRemove}>x</button>
        </div>);

    if(edit){
        jsx = (
            <div>
                <li key={id}>{task}</li>
                <button onClick={handleUpdate}>Update</button>
                <button onClick={handleRemove}>x</button>
                <UpdateTodoForm update={update} id={id} tasks={task}/>
            </div>)
    }
    
    

    return jsx;
}

export default Todo;