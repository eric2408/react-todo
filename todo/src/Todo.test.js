import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

it("renders without crashing", function() {
  render(<Todo />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});


it('runs edit function on form submit', function(){
    const mock = jest.fn();
    const {getByText} = render(<Todo update={mock}/>);
    fireEvent.click(getByText('Update'));
    fireEvent.click(getByText('Update Todo'));
    expect(mock).toHaveBeenCalled();
});

it('runs delete function when x button is clicked', function(){
    const mock = jest.fn();
    const {getByText} = render(<Todo remove={mock}/>);
    fireEvent.click(getByText("x"));
    expect(mock).toHaveBeenCalled();
});