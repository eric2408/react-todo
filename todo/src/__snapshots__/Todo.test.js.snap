// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <li>
      empty task
    </li>
    <button>
      Update
    </button>
    <button>
      x
    </button>
  </div>
</DocumentFragment>
`;
